File: Core/Entities/Invoice.cs

namespace Core.Entities
{
    public class Invoice : BaseEntity
    {
        public decimal TotalAmount { get; set; }
        public DateTime Date { get; set; }
        public string PaymentType { get; set; }

        public int CustomerId { get; set; }
        public Customer Customer { get; set; }

        public int ReservationId { get; set; }
        public Reservation Reservation { get; set; }
    }
}

File: Client/src/app/shared/models/invoice.ts
export interface Invoice {
    id: number
    totalAmount: number
    date: string
    paymentType: string
    customer: string
    reservation: string
}

export interface IInvoiceToCreate {
    totalAmount: number
    date: string
    paymentType: string
    customerId: number
    reservationId: number
}

export class InvoiceFormValues implements IInvoiceToCreate {
    totalAmount = 0;
    date = '';
    paymentType = ''
    customerId!: number
    reservationId!: number

    constructor(init?: InvoiceFormValues) {
        Object.assign(this, init);
    }
}